From 0dddf385b6b9722039ebc224fde651dd1964464f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Krause?= <joerg.krause@embedded.rocks>
Date: Sun, 12 Mar 2017 09:21:10 +0100
Subject: [PATCH 1/2] Add option to disable building the examples
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Krause <joerg.krause@embedded.rocks>
---
 configure.ac    |  8 ++++++++
 src/Makefile.am | 11 ++++++++---
 2 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/asio/configure.ac b/asio/configure.ac
index d453602e..360a371d 100644
--- a/asio/configure.ac
+++ b/asio/configure.ac
@@ -161,6 +161,12 @@ AC_ARG_ENABLE(separate-compilation,
   SEPARATE_COMPILATION=yes
 ])
 
+AC_ARG_ENABLE(examples,
+[  --disable-examples  disable the example programs],
+[
+  ENABLE_EXAMPLES=no
+])
+
 AC_ARG_ENABLE(boost-coroutine,
 [  --enable-boost-coroutine  use Boost.Coroutine to implement stackful coroutines],
 [
@@ -252,6 +258,8 @@ AM_CONDITIONAL(STANDALONE,test x$STANDALONE = xyes)
 
 AM_CONDITIONAL(SEPARATE_COMPILATION,test x$SEPARATE_COMPILATION = xyes)
 
+AM_CONDITIONAL(ENABLE_EXAMPLES,test x$ENABLE_EXAMPLES != xno)
+
 AM_CONDITIONAL(HAVE_BOOST_COROUTINE,test x$HAVE_BOOST_COROUTINE = xyes)
 
 AM_CONDITIONAL(WINDOWS_TARGET,test x$WINDOWS != xno)
diff --git a/asio/src/Makefile.am b/asio/src/Makefile.am
index 48c3eac7..f6530a09 100644
--- a/asio/src/Makefile.am
+++ b/asio/src/Makefile.am
@@ -1,10 +1,15 @@
+if ENABLE_EXAMPLES
+EXAMPLES_CPP03 = examples/cpp03
+EXAMPLES_CPP11 = examples/cpp11
+endif
+
 if STANDALONE
-SUBDIRS = examples/cpp11 tests
+SUBDIRS = $(EXAMPLES_CPP11) tests
 else
-SUBDIRS = examples/cpp03 tests
+SUBDIRS = $(EXAMPLES_CPP03) tests
 endif
 
-DIST_SUBDIRS = examples/cpp03 examples/cpp11 tests
+DIST_SUBDIRS = $(EXAMPLES_CPP03) $(EXAMPLES_CPP11) tests
 
 EXTRA_DIST = \
 	Makefile.mgw \
-- 
2.12.0
